src/widget/weffectselector.cpp

```diff
@@ -0,0 +1,94 @@
+#include <QtDebug>
+
+#include "widget/weffectselector.h"
+
+#include "effects/effectsmanager.h"
+#include "widget/effectwidgetutils.h"
+
+namespace {
+    const QString nullEffectId = "NULL";
+}
+
+WEffectSelector::WEffectSelector(QWidget* pParent, EffectsManager* pEffectsManager)
+        : QComboBox(pParent),
+          WBaseWidget(pParent),
+          m_pEffectsManager(pEffectsManager) {
+
+    setInsertPolicy(QComboBox::InsertAlphabetically);
+    // TODO(xxx): filter out blacklisted effects
+    // https://bugs.launchpad.net/mixxx/+bug/1653140
+    QList<QPair<QString,QString> > idNamePairs = m_pEffectsManager->getEffectShortNamesFiltered(nullptr);
+    for (const auto& effectPair : idNamePairs) {
+        addItem(effectPair.second, QVariant(effectPair.first));
+    }
+    //: Displayed when no effect is loaded
+    addItem(tr("None"), QVariant(nullEffectId));
```

Comment:
Instead of having a null string constant, you can use the fact the QVariant and QString both have ```isNull``` methods for when their no-args constructor is used. 

```QString("").isNull() == false``` and ```QString().isNull() == true```

You can use ```QVariant()``` and then check ```QVariant::isNull``` instead of